static/app/views/profiling/profileDetails.tsx

```diff
@@ -149,6 +148,127 @@ function ProfileDetails() {
     [location, search]
   );
 
+  const [filters, setFilters] = useState<Partial<Record<TableColumnKey, string[]>>>({});
+
+  const columnFilters = useMemo(() => {
+    function makeOnFilterChange(key: string) {
+      return values => {
+        setFilters(prevFilters => ({
+          ...prevFilters,
+          [key]: values.length > 0 ? values.map(val => val.value) : undefined,
+        }));
+      };
+    }
+    return {
+      type: {
+        values: ['application', 'system'],
+        onChange: makeOnFilterChange('type'),
+      },
+      image: {
+        values: pluckUniqueValues(slowestFunctions, 'image').sort((a, b) =>
+          a.localeCompare(b)
+        ),
+        onChange: makeOnFilterChange('image'),
+      },
+    };
+  }, [slowestFunctions]);
+
+  const currentSort = useMemo<GridColumnSortBy<TableColumnKey>>(() => {
+    let key = location.query?.functionsSort ?? '';
+
+    const defaultSort = {
+      key: 'self weight',
+      order: 'desc',
+    } as GridColumnSortBy<TableColumnKey>;
+
+    const isDesc = key[0] === '-';
+    if (isDesc) {
+      key = key.slice(1);
+    }
+
+    if (!key || !tableColumnKey.includes(key as TableColumnKey)) {
+      return defaultSort;
+    }
+
+    return {
+      key,
+      order: isDesc ? 'desc' : 'asc',
+    } as GridColumnSortBy<TableColumnKey>;
+  }, [location.query]);
+
+  useEffect(() => {
+    const removeListener = browserHistory.listenBefore((nextLocation, next) => {
+      if (location.pathname === nextLocation.pathname) {
+        next(nextLocation);
+        return;
+      }
+
+      if ('functionsSort' in nextLocation.query) {
+        delete nextLocation.query.functionsSort;
+      }
+
+      next(nextLocation);
+    });
+
+    return removeListener;
+  });
+
+  const generateSortLink = useCallback(
+    (column: TableColumnKey) => {
+      if (!SORTABLE_COLUMNS.has(column)) {
+        return () => undefined;
```

Comment:
Would throwing an error make more sense here? We should not not allow user action on something that we know beforehand will result in any changes in order to avoid user confusion.

Reply:
would be; but ideally we just wouldn't have the button clickable if it wasn't defined as sortable. which is actually how it works. if you follow the code, we end up in a place where we never reach this code path;


![image](https://user-images.githubusercontent.com/7349258/196195304-5ca15ffd-e61b-477b-960e-ffda6cc23d04.png)

![image](https://user-images.githubusercontent.com/7349258/196196342-5cf66a58-8e9c-4904-8091-8c2bb6c4be6d.png)


this is a weird nuance of the grid api, needs some work.